android-o/AutofillFramework/AutofillFramework/app/CountdownTimerImpl.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using Android.OS;
+using Android.Widget;
+
+namespace AutofillFramework.app
+{
+	public class CountdownTimerImpl : CountDownTimer
+	{
+		public TextView CountdownText { get; set; }
+		public WelcomeActivity Activity { get; set; }
+
+		public CountdownTimerImpl(WelcomeActivity activity, TextView countdownText,
+			long millisInFuture, long countDownInterval) : base(millisInFuture, countDownInterval) 
+			{
+			Activity = activity;
+			CountdownText = countdownText;
+		}
+
+		public override void OnFinish()
+		{
+			if (!Activity.IsFinishing)
+			{
+				Activity.Finish();
+			}
+		}
+
+		public override void OnTick(long millisUntilFinished)
+		{
+			int secondsRemaining = Java.Lang.Math.ToIntExact(millisUntilFinished / 1000);
```

Comment:
I don't think we need the Java.Lang.Math call here.

Reply:
Done